src/Worker/Worker.php

```diff
@@ -82,6 +83,7 @@ public function start(): void
             } finally {
                 if ($this->kernel instanceof RebootableInterface && $this->dependencies->getKernelRebootStrategy()->shouldReboot()) {
                     $this->kernel->reboot(null);
+                    $this->eventDispatcher->dispatch(new WorkerKernelRebootedEvent());
```

Comment:
`$this-eventDispatcher` contains the dispatcher created on the first kernel boot.

During a reboot, the dependency injection container is destroyed and recreated.

This is why I created the `Dependencies` class, the dispatcher should be added to it, so that the listeners have fresh dependencies.

Reply:
Good point, I guess that goes for the other calls as well so I removed the `Worker::$eventDispatcher` completely

Reply:
Good catch for the other calls.

Removing the argument of the constructor is a BC break, and I marked the class as internal only in 1.3.0 so I prefer to keep the constructor signature untouched. No problem to remove the property.

Reply:
👍 Its back now :)